homeassistant/components/homekit/__init__.py

```diff
@@ -134,28 +120,87 @@
 )
 
 
-async def async_setup(hass, config):
-    """Set up the HomeKit component."""
-    _LOGGER.debug("Begin setup HomeKit")
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the HomeKit from yaml."""
 
-    aid_storage = hass.data[AID_STORAGE] = AccessoryAidStorage(hass)
-    await aid_storage.async_initialize()
+    hass.data.setdefault(DOMAIN, {})
 
-    hass.http.register_view(HomeKitPairingQRView)
+    _async_register_events_and_services(hass)
+
+    if not config:
+        return True
+
+    current_entries = hass.config_entries.async_entries(DOMAIN)
+
+    conf = config.get(DOMAIN)
+
+    for entry in current_entries:
+        if entry.source == SOURCE_IMPORT:
+            # If they alter the yaml config we import the changes
+            # since there currently is no practical way to support
+            # all the options in the UI at this time.
+            data = conf.copy()
+            options = {}
+            for key in CONFIG_OPTIONS:
+                options[key] = data[key]
+                del data[key]
+
+            hass.config_entries.async_update_entry(entry, data=data, options=options)
+            return True
+
+    hass.async_create_task(
```

Comment:
Here, I think you can make multiple configurations through YAML be taken into consideration. Like this:

```python
for conf in config[DOMAIN]:
        hass.async_create_task(
            hass.config_entries.flow.async_init(
                DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
            )
        )
```

Of course, you'd also need to change the config schema to allow this.

Reply:
@joogps I didn't do this in order to avoid making this a breaking change.

Reply:
Ohh, ok. Didn't think about that 😅

Reply:
But would it affect users? I think (and I'm probably wrong) that single configurations (without dashes) would still work fine.

Reply:
We would need to migrate the aid storage and the homekit state files as well. Could do this a few versions from now once everything has already been imported so only people who didn't upgrade though the versions would loose their custom names and room assignments.

Reply:
I'll see if I can come up with a way to support multiple yaml entries without making this a breaking change.  Unfortunately that will make this PR quite a bit larger